include/dxc/Support/HLSLOptimizationOptions.h

```diff
@@ -0,0 +1,24 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// HLSLOptions.h                                                             //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Option defined related to optimization customization.                     //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#ifndef LLVM_HLSL_OPTIMIZATION_OPTIONS_H
+#define LLVM_HLSL_OPTIMIZATION_OPTIONS_H
+
+namespace hlsl {
+  // Optimizations that can be disabled
+  enum OptimizationToggles {
```

Comment:
I think I'd really prefer this to be a struct with bitfields that can be used directly wherever we set or read these options.  Additionally, I'd prefer not having something called Toggle, since that doesn't really say what it does (does it enable or disable?).  Let's just make it like the option we set on the command line for now.
Like:
```
struct OptimizationOptions {
  unsigned DisableGVN : 1;
  unsigned Reserved : 31;
};
```

Reply:
I was worried about name clashes, but with the struct that won't be a problem.

Reply:
The problem with the struct is that it will have to be visible to every place this gets copied, but with the new header that will be less of a problem.